spyder/plugins/editor/confpage.py

```diff
@@ -148,6 +148,21 @@ def setup_page(self):
             'tab_stop_width_spaces',
             default=4, min_=1, max_=8, step=1)
 
+        format_on_save = CONF.get(
+            'completions',
+            ('provider_configuration', 'lsp', 'values', 'format_on_save'),
+            False
+        )
+        if format_on_save:
```

Comment:
This is ok when preferences is opened and auto-format on save is already set, but not if you enable it and go to this tab in the editor. In that case, these options are still enabled (but they shouldn't).

To solve that, perhaps you could use `SpyderConfigurationObserver` here?

Reply:
I think that even with that the config page will not take into account tha change unless the user clicks at leats the Apply button, is that ok?

Reply:
Checking the usage of `SpyderConfigurationObserver` for some reason it gets stuck :/ Could we merge this as it is (to have the translation string to do the translation PR update) and check the missing UX behavior in a different following PR @ccordoba12 ?